docs/DeveloperGuide.md

```diff
@@ -1,60 +1,69 @@
+The `TextUi` class serves strictly as intermediary between the user and the program, and does not directly interact with any components other than the `Main` class. It fulfils the following roles:
+
+- Listens for and grabs the user's input using a `Scanner`, and returns it to `Main` as a string for further processing.
+- Displays any command results or status and error messages to the user.
 
 ### Parser Component
+
 ![Class Diagram](http://www.plantuml.com/plantuml/proxy?src=https://raw.githubusercontent.com/AY2122S2-CS2113T-T10-3/tp/master/docs/Parser.puml)
-How the parsing works:
-1. All XYZParser(XYZ is a placeholder e.g.  AddParser) and ModHappyParser interited from Parser class, which defines parseString() method that can parse based on well defined command Regular expression(Regex) and command groups. XYZParser parses user input in multiple-layer manner to suport complex and extendable commands.
-2. When called upon to parse a user command input, the ModHappyParser class, which serves as a general singleton parser of Mod Happy,  creates an XYZParser
-3. XYZParser further parses the user input and will return corresponding Command objects.
+
+The `Parser` component serves to interpret user input and construct the relevant `Command` objects to be returned to `Main` for execution later on. This component comprises the following classes:
+
+* `Parser`, which is an abstract class serving as the parent of all other classes in this component. This class implements basic parsing functionality through the use of regular expressions and the built-in `Pattern` class. This functionality is modified by its child classes to suit their own purposes.
+* `ModHappyParser`, which identifies the command word present in the user input and invokes the relevant command-specific parser.
+* A variety of command-specific parsers (e.g. `AddParser` for the `add` command), referred to in this guide as `XYZParser` for simplicity. These classes perform further parsing on any command-specific arguments, constructing and returning the corresponding `XYZCommand` object.
+
+> 📔 <span style="color:#00bb00">**NOTE:**</span>
+> 
+> `NoArgumentParser` is an exception to the above; instead of being associated with a single command type, it is responsible for handling all commands which do not accept any arguments.
+
+The following details how the `Parser` component works at runtime:
+
+1. A single `ModHappyParser` instance is initialised by `Main` during at the start of the program.
+2. Each time the user inputs a command, `ModHappyParser`'s `parseCommand()` method with the input as the parameter.
+3. `ModHappyParser` identifies the relevant command-specific parser `XYZParser` and passes on the remaining unparsed arguments to its `parseCommand()` method.
+4. `XYZParser` parses the remaining command arguments and uses them to construct an `XYZCommand` instance, which is subsequently returned to `Main`.
+
```

Comment:
Should this be better demonstrated with a sequence diagram?

Reply:
We can add a sequence diagram to showcase this later on but probably in a future pull request